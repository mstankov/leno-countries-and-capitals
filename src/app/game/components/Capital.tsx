import { Box } from "@mui/material";
import { FC } from "react";
import { Capital as CapitalType } from "src/api/types";

export type CapitalProps = {
  capital: CapitalType;
  onClick: (capital: CapitalType) => void;
};

const Capital: FC<CapitalProps> = ({ capital, onClick }) => {
  const onClickHandler = () => {
    onClick(capital);
  };

  return (
    <Box data-testid={`capital-${capital}`} onClick={onClickHandler}>
      {capital}
    </Box>
  );
};

export default Capital;
