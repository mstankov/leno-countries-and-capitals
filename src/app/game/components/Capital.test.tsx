import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Capital from "./Capital";

describe(Capital.name, () => {
  it("should render", () => {
    const { container } = render(
      <Capital capital="Amsterdam" onClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should call callback on click with capital name", async () => {
    userEvent.setup();

    const onClick = jest.fn();
    render(<Capital capital="Amsterdam" onClick={onClick} />);

    await userEvent.click(await screen.findByTestId("capital-Amsterdam"));

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith("Amsterdam");
  });
});
