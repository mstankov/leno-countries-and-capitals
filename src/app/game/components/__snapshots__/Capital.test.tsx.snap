// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capital should render 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="capital-Amsterdam"
  >
    Amsterdam
  </div>
</div>
`;
